modal/client.py

```diff
@@ -75,46 +77,44 @@ async def _http_check(url: str, timeout: float) -> str:
 
 async def _grpc_exc_string(exc: GRPCError, method_name: str, server_url: str, timeout: float) -> str:
     http_status = await _http_check(server_url, timeout=timeout)
-    return f"{method_name}: {exc.message} [GRPC status: {exc.status.name}, {http_status}]"
+    return f"{method_name}: {exc.message} [gRPC status: {exc.status.name}, {http_status}]"
 
 
 class _Client:
     _client_from_env = None
     _client_from_env_lock = None
 
-    client_type: int
-
     def __init__(
         self,
-        server_url,
-        client_type,
-        credentials,
-        version=__version__,
-        *,
-        no_verify=False,
+        server_url: str,
+        client_type: int,
+        credentials: Optional[Tuple[str, str]],
+        version: str = __version__,
     ):
         """The Modal client object is not intended to be instantiated directly by users."""
         self.server_url = server_url
         self.client_type = client_type
         self.credentials = credentials
         self.version = version
-        self.no_verify = no_verify
+        self._authenticated = False
         self._pre_stop: Optional[Callable[[], Awaitable[None]]] = None
-        self._channel = None
+        self._channel: Optional[grpclib.client.Channel] = None
         self._stub: Optional[api_grpc.ModalClientStub] = None
 
     @property
     def stub(self) -> Optional[api_grpc.ModalClientStub]:
         """mdmd:hidden"""
         return self._stub
 
+    @property
+    def authenticated(self) -> bool:
+        """mdmd:hidden"""
+        return getattr(self, "_authenticated", False)
```

Comment:
I think this should work here

```suggestion
        return self._authenticated
```